hoot-core-test/src/test/cpp/hoot/core/io/OsmApiWriterTest.cpp

```diff
@@ -244,18 +258,22 @@ class OsmApiWriterTest : public HootTestFixture
         "</osmChange>\n",
         writer.getFailedChangeset());
     }
+#endif
   }
 
   void runRetryConflictsTest()
   {
-    //  Suppress the OsmApiWriter errors by temporarily changing the log level.
-    //  We expect the all of the errors
+#ifdef RUN_LOCAL_TEST_SERVER
+    //  Suppress the OsmApiWriter errors by temporarily changing the log level
+    //  when the log level is Info or above because we expect the all of the errors.
+    //  Below Info is Debug and Trace, those are set because we want to see everything
     Log::WarningLevel logLevel = Log::getInstance().getLevel();
-    Log::getInstance().setLevel(Log::Fatal);
+    if (Log::getInstance().getLevel() >= Log::Info)
+      Log::getInstance().setLevel(Log::Fatal);
```

Comment:
This is a change I made a few different places, turning off logging almost completely is great when we don't want to see errors that are expected but using `--debug` or `--trace` for a single test doesn't output the expected trace and debug statements because the log has turned them off.  This `if` fixes that issue.  I looked through all of the tests in the code base and update a few of them that exhibited the same problem.

Reply:
Nice!